File: fundamentals/3_input-output/InputOutput_CPP/InputOutput.cpp
/**
 * Declares some variables related to the details of a particular individual.
 * This includes name, surname, age, height, weight, gender, and marital status.
 * Then prompts the user to enter the values for these variables.
 * Then displays this information on the console.
 */
#include <iostream>
#include <string>

int main()
{
	// Variable Declarations
	std::string name = "", surname = "";
	int age;
	double height, weight;
	char gender;
	bool married;

	//User Input
	std::cout << "Name: ";
	std::cin >> name;

	std::cout << "Surname: ";
	std::cin >> surname;

	std::cout << "Age: ";
	std::cin >> age;

	std::cout << "Height (cm): ";
	std::cin >> height;

	std::cout << "Weight (kg): ";
	std::cin >> weight;

	std::cout << "Gender (M for Male | F for Female): ";
	std::cin.get();
	std::cin >> gender;

	std::cout << "Married (1 for Yes | 0 for No): ";
	std::cin.get();
	std::cin >> married;

	// Displaying the Information
	std::cout << "\n--- Personal Details ---\n";
	std::cout << "Hello " << name << ", following are your personal details:" << std::endl;
	std::cout << "\tFull Name: " << name << " " << surname << std::endl;
	std::cout << "\tAge: " << age << std::endl;
	std::cout << "\tHeight: " << height << std::endl;
	std::cout << "\tWeight: " << weight << std::endl;
	std::cout << "\tGender: " << ((gender == 'm' || gender == 'M') ? "Male" : "Female") << std::endl;
	std::cout << "\tMarital Status: " << (married ? "Married" : "Not Married") << std::endl;

	return 0;
}

File: fundamentals/3_input-output/InputOutput_JAVA/InputOutput.java
/*
 * Declares some variables related to the details of a particular individual.
 * This includes name, surname, age, height, weight, gender, and marital status.
 * Then prompts the user to enter the values for these variables.
 * Then displays this information on the console.
 */

import java.util.Scanner;

public class InputOutput {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);	// Declares an input scanner to read from the console

		// Variable Declarations
		String name = "", surname = "";
		int age;
		double height, weight;
		char gender;
		boolean married;

		// User Input
		System.out.print("Name: ");
		name = scanner.nextLine();

		System.out.print("Surname: ");
		surname = scanner.nextLine();

		System.out.print("Age: ");
		age = scanner.nextInt();

		System.out.print("Height (cm): ");
		height = scanner.nextDouble();

		System.out.print("Weight (kg): ");
		weight = scanner.nextDouble();

		System.out.print("Gender (M for Male | F for Female): ");
		gender = scanner.next().charAt(0);

		System.out.print("Married (true | false): ");
		married = scanner.nextBoolean();

		// Output to Console
		System.out.println("\n--- PERSONAL DETAILS ---");
		System.out.println("Hello " + name + ", following are your personal details:");
		System.out.println("\tFull Name: " + name + " " + surname);
		System.out.println("\tAge: " + age);
		System.out.println("\tHeight: " + height + " cm");
		System.out.println("\tWeight: " + weight + " kg");
		System.out.println("\tGender: " + ((gender == 'm' || gender == 'M') ? "Male" : "Female"));
		System.out.println("\tMarital Status: " + (married ? "Married" : "Not Married"));

		scanner.close();	// Close the scanner to prevent resource leaks
	}
}